meta-resin-common/recipes-bsp/u-boot/patches/resin-specific-env-configuration.patch

```diff
@@ -1,48 +1,53 @@
+From 20c288af1e401549f434f22e420dfa1409c57078 Mon Sep 17 00:00:00 2001
 From: Andrei Gherzan <andrei@resin.io>
 Date: Wed, 4 Jan 2017 15:58:42 +0100
-Subject: Add resin specific environment configuration
+Subject: [PATCH] Add resin specific environment configuration
 
 This patch defines all the needed bits on which resinhup relies when swiching
-the root partition. The interface with uboot is done through a file called
-uEnv.txt located in the first partition which, if it exists, might overwrite the
-default root partition index (by default 2). In order to overwrite this,
-uEnv.txt must define `resin_root_part`.
+the root partition. The interface with uboot is done through an env file
+located in the first partition which, if it exists, might overwrite the default
+root partition index. In order to overwrite this, the env file must define
+`resin_root_part`.
 
 Signed-off-by: Andrei Gherzan <andrei@resin.io>
 
 Upstream-Status: Inappropriate [configuration]
-
 ---
- include/env_resin.h | 40 ++++++++++++++++++++++++++++++++++++++++
- 1 file changed, 40 insertions(+)
+ include/env_resin.h | 62 +++++++++++++++++++++++++++++++++++++++++++++++++++++
+ 1 file changed, 62 insertions(+)
  create mode 100644 include/env_resin.h
 
 diff --git a/include/env_resin.h b/include/env_resin.h
 new file mode 100644
-index 0000000..ef8a2b0
+index 0000000..dfef80c
 --- /dev/null
 +++ b/include/env_resin.h
-@@ -0,0 +1,39 @@
+@@ -0,0 +1,61 @@
 +#ifndef HEADER_ENV_RESIN_H
 +#define HEADER_ENV_RESIN_H
 +
 +/*
 + * Assumes defined:
-+ *		resin_kernel_load_addr - kernel load address as we use the same to load the env file
-+ *		resin_mmcdev - current MMC device to use
-+ *		resin_root_part - existing in the env file to import
++ *     resin_kernel_load_addr - kernel load address as we use the same
++ *                              to load the env file
++ *     resin_root_part        - existing in the env file to import (optional)
 + * Defines:
-+ *		resin_set_kernel_root - needs to be integrated with board specific configuration
-+ *		resin_kernel_root - the root kernel argument 
++ *     resin_set_kernel_root  - needs to be integrated with board
++ *                              specific configuration
++ *     resin_kernel_root      - the root kernel argument
++ *     resin_mmcdev          - MMC device index to be used at boot
 + */
++
++#include <config_resin.h>
++
 +#define RESIN_ENV \
-+	"resin_env_file=uEnv.txt\0" \
-+	"resin_boot_part=1\0" \
-+	"resin_root_part=2\0" \
++	"resin_env_file=" __stringify(RESIN_ENV_FILE) "\0" \
++	"resin_flasher_external_flag_file=" __stringify(RESIN_FLASHER_EXTERNAL_FLAG_FILE) "\0" \
++	"resin_boot_part=" __stringify(RESIN_BOOT_PART) "\0" \
++	"resin_root_part=" __stringify(RESIN_DEFAULT_ROOT_PART) "\0" \
 +	\
 +	"resin_find_root_part_uuid=" \
-+		"part uuid mmc ${resin_mmcdev}:${resin_root_part} root_part_uuid\0" \
++		"part uuid mmc ${resin_mmcdev}:${resin_root_part} resin_root_part_uuid\0" \
```

Comment:
This will need to be enabled as part of the board specific patch for every board. Please disregard the above comment as it's not actually related to this PR